src/akkudoktoreos/server/fastapi_server.py

```diff
@@ -106,6 +106,25 @@ def fastapi_strompreis() -> list[float]:
     specific_date_prices = marketprice_series.loc[
         prediction_eos.start_datetime : prediction_eos.end_datetime
     ]
+    # TODO merge
```

Comment:
@b0661 I need help here from merging the latest changes from main. I don't know how to proceed here (add charges to electricity provider, provide mean, ...) See changes to src/akkudoktoreos/prediction/price_forecast.py

Reply:
Price forecast using akkudoktor.net is in src/akkudoktoreos/prediction/elecpriceakkudoktor.py. The adaptions have to be made here:

```Python
        # Assure price ends at end_time
        if compare_datetimes(self[-1].date_time, self.end_datetime).lt:
            record = ElecPriceDataRecord(
                date_time=self.end_datetime,
                elecprice_marketprice=self[-1].elecprice_marketprice,
            )
            self.append(record)
```

As akkudoktor.net currently only provides 24 prices in 1 hour intervals. Currently the last price available from akkudoktor.net is set also on 'self.end_datetime'. When retrieving a price array the missing prices in between will be interpolated using the `key_to_array()` function.

To add Andreas' changes you have to add the missing prices here (see above).  I think he made some calculations to get the missing 24 prices using the data retrieved from akkudoktor.net. Currently not enough past data is retrieved from akkudoktor net and only `self.config.prediction_historic_hours` (typically 24 hours) past data is stored in the prediction data records.

```Python
    @cache_in_file(with_ttl="1 hour")
    def _request_forecast(self) -> AkkudoktorElecPrice:
        """Fetch electricity price forecast data from Akkudoktor API.

        This method sends a request to Akkudoktor's API to retrieve forecast data for a specified
        date range. The response data is parsed and returned as JSON for further processing.

        Returns:
            dict: The parsed JSON response from Akkudoktor API containing forecast data.

        Raises:
            ValueError: If the API response does not include expected `electricity price` data.
        """
        source = "https://api.akkudoktor.net"
        date = to_datetime(self.start_datetime, as_string="Y-M-D")
        last_date = to_datetime(self.end_datetime, as_string="Y-M-D")
        response = requests.get(
            f"{source}/prices?date={date}&last_date={last_date}&tz={self.config.timezone}"
        )
        response.raise_for_status()  # Raise an error for bad responses
        logger.debug(f"Response from {source}: {response}")
        akkudoktor_data = self._validate_data(response.content)
        # We are working on fresh data (no cache), report update time
        self.update_datetime = to_datetime(in_timezone=self.config.timezone)
        return akkudoktor_data
```

I assume `date` has to be changed to be ??7?? days before `self.start_datetime`. Then you get all the data to do the extra calculations. The data will be in `akkudoktor_data`, which you can use to do the extra calculations.

```Python
    from akkudoktoreos.utils.datetimeutils import to_duration
    date = to_datetime(self.start_datetime  - to_duration("7 days"), as_string="Y-M-D")
```

Reply:
Will add solution to next config/ prediction update/fix.